mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -342,23 +343,24 @@ def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
             file_size = os.path.getsize(local_file)
             num_chunks = _compute_num_chunks(local_file, _MULTIPART_UPLOAD_CHUNK_SIZE)
             use_single_part_upload = num_chunks == 1
-            for index in range(num_chunks):
-                start_byte = index * _MULTIPART_UPLOAD_CHUNK_SIZE
-                future = self.thread_pool.submit(
-                    self._retryable_adls_function,
-                    func=patch_adls_file_upload,
-                    artifact_path=artifact_path,
-                    sas_url=credentials.signed_uri,
-                    local_file=local_file,
-                    start_byte=start_byte,
-                    size=_MULTIPART_UPLOAD_CHUNK_SIZE,
-                    position=start_byte,
-                    headers=headers,
-                    is_single=use_single_part_upload,
-                )
-                futures[future] = index
+            with self._create_thread_pool() as executor:
```

Comment:
Can we insert inline comments explaining why we're creating new pools here and link to the deadlock article you sent?